add in MCif Delete ,authorization dan Mapping builder

Yaksyin opened the pull request:
add in TSavingAccount Put/Edit update
add MCif Delete , authorization dan  maping build
![image_2024-12-14_051901540](https://github.com/user-attachments/assets/a5581a62-c21a-4a66-965a-fb9a9c2c94dc)
![image](https://github.com/user-attachments/assets/eee7fc8f-7e70-4485-981e-1d540536ce68)
![image](https://github.com/user-attachments/assets/4d67ca51-2c0e-45a0-b0fa-be37dce088b6)

{
    "accountNumber": "4444567890",
    "cifId": "1820741",
    "savingId": "431213"
}
RESPONSE :
{
    "savingAccountId": "97149824619284",
    "accountNumber": "4444567890",
    "cifId": "1820741",
    "beginBalance": 0.00,
    "endBalance": 0.00,
    "currentBalance": 0.00,
    "statusId": "1",
    "statusName": "Pending",
    "authorizationAt": null,
    "authorizationBy": "",
    "savingId": "expresi",
    "savingName": "expresi",
    "deleted": false
}

Yaksyin committed:
add in MCif Delete ,authorization dan Mapping builder
add in TSavingAccount Put/Edit update

xuibdg commented on src/main/java/com/b2camp/simple_core_banking/controller/TSavingAccountController.java:
```diff
@@ -1,16 +1,23 @@
     }
+
+    @PutMapping("/{savingAccountId}")
+    public TSavingAccountResponse createTSavingAccount(@PathVariable String savingAccountId,
```
so is this controller for create or edit?

Yaksyin committed:
Merge remote-tracking branch 'origin/develop' into cif

# Conflicts:
#	src/main/java/com/b2camp/simple_core_banking/service/impl/MUserServiceImpl.java

Yaksyin committed:
add in MCif Delete ,authorization dan Mapping builder
add in TSavingAccount Put/Edit update ADD transactinal ,propagation and isolitation

Yaksyin committed:
add in MCif Delete ,authorization dan Mapping builder
add in TSavingAccount Put/Edit update ADD transactinal ,propagation and isolitation

Yaksyin committed:
Merge remote-tracking branch 'refs/remotes/origin/develop' into CifDelete

# Conflicts:
#	src/main/java/com/b2camp/simple_core_banking/controller/MCifController.java
#	src/main/java/com/b2camp/simple_core_banking/controller/TSavingAccountController.java
#	src/main/java/com/b2camp/simple_core_banking/dto/MCifResponse.java
#	src/main/java/com/b2camp/simple_core_banking/dto/TSavingAccountRequest.java
#	src/main/java/com/b2camp/simple_core_banking/dto/TSavingAccountResponse.java
#	src/main/java/com/b2camp/simple_core_banking/entity/TSavingAccount.java
#	src/main/java/com/b2camp/simple_core_banking/repository/MCifRepository.java
#	src/main/java/com/b2camp/simple_core_banking/service/MCifService.java
#	src/main/java/com/b2camp/simple_core_banking/service/TSavingAccountService.java
#	src/main/java/com/b2camp/simple_core_banking/service/impl/MCifServiceImpl.java
#	src/main/java/com/b2camp/simple_core_banking/service/impl/TSavingAccountServiceImpl.java

Yaksyin committed:
Merge remote-tracking branch 'origin/develop' into CifDelete

# Conflicts:
#	src/main/java/com/b2camp/simple_core_banking/dto/MCifResponse.java
#	src/main/java/com/b2camp/simple_core_banking/dto/TSavingAccountResponse.java
#	src/main/java/com/b2camp/simple_core_banking/entity/TSavingAccount.java

xuibdg commented on src/main/java/com/b2camp/simple_core_banking/controller/MCifController.java:
```diff
@@ -1,9 +1,9 @@
 import com.b2camp.simple_core_banking.service.MCifService;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
```
create single import

xuibdg commented on src/main/java/com/b2camp/simple_core_banking/dto/MCifResponse.java:
```diff
@@ -23,10 +23,7 @@ public class MCifResponse {
-
-        return cifId;
-    }
+    public String getCifId() { return cifId; }
```
kenapa ini masih pake getter setter manual? bukannya di develop sudah pake lombok?

xuibdg commented on src/main/java/com/b2camp/simple_core_banking/dto/TSavingAccountResponse.java:
```diff
@@ -19,6 +20,15 @@ public class TSavingAccountResponse {
     //
 
+
+    public String getStatusName() {
```
kenapa ini masih pake getter setter manual? bukannya di develop sudah pake lombok?

xuibdg commented on src/main/java/com/b2camp/simple_core_banking/service/TSavingAccountService.java:
```diff
@@ -1,4 +1,26 @@
+
+    Optional<TSavingAccountResponse> findBySavingAccountId(String savingAccountId);
+
+    @Transactional(propagation = Propagation.REQUIRED,rollbackFor = Exception.class)
```
ini kenapa? @Transactional di simpen di service?

xuibdg commented on src/main/java/com/b2camp/simple_core_banking/service/impl/TSavingAccountServiceImpl.java:
```diff
@@ -1,6 +1,255 @@
+    @Transactional (propagation = Propagation.REQUIRED,rollbackFor = Exception.class)
+    public TSavingAccountResponse updateSavingAccount(String savingAccountId, TSavingAccountRequest request) {
+        TSavingAccount tSavingAccount = tSavingAccountRepository
+                .findById(savingAccountId).orElse(null);
```
ini udah mencari data berdasarkan isDeleted is false?

xuibdg commented on untitled/pom.xml:
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
kenapa menambahkan untitled/pom.xml?

Yaksyin committed:
resolve conflick and fixing TSavingAccount Update and Mcif Delete , Authorizartion

Yaksyin committed:
resolve conflick and fixing TSavingAccount Update and Mcif Delete , Authorizartion

Yaksyin committed:
resolve conflick and fixing TSavingAccount Update and Mcif Delete , Authorizartion
